Sourcecode/private/mx/impl/PropertiesWriter.cpp

```diff
@@ -165,26 +169,59 @@ namespace mx
             }
             
-            key->getKeyChoice()->setChoice( core::KeyChoice::Choice::traditionalKey );
-            auto traditionalKey = key->getKeyChoice()->getTraditionalKey();
-            traditionalKey->getFifths()->setValue( core::FifthsValue{ inKeyData.fifths } );
-            
-            if ( inKeyData.cancel != 0 )
+            if (!inKeyData.nonTraditionalKey)
             {
-                traditionalKey->setHasCancel( true );
-                traditionalKey->getCancel()->setValue( core::FifthsValue{ inKeyData.cancel } );
-            }
-            
-            if( inKeyData.mode == api::KeyMode::major || inKeyData.mode == api::KeyMode::minor )
-            {
-                traditionalKey->setHasMode( true );
-                traditionalKey->getMode()->setValue( core::ModeValue{ inKeyData.mode == api::KeyMode::major ? core::ModeEnum::major : core::ModeEnum::minor } );
+                key->getKeyChoice()->setChoice(core::KeyChoice::Choice::traditionalKey);
+                auto traditionalKey = key->getKeyChoice()->getTraditionalKey();
+                traditionalKey->getFifths()->setValue(core::FifthsValue{ inKeyData.fifths });
+
+                if (inKeyData.cancel != 0)
+                {
+                    traditionalKey->setHasCancel(true);
+                    traditionalKey->getCancel()->setValue(core::FifthsValue{ inKeyData.cancel });
+                }
+
+                if (inKeyData.mode == api::KeyMode::major || inKeyData.mode == api::KeyMode::minor)
+                {
+                    traditionalKey->setHasMode(true);
+                    traditionalKey->getMode()->setValue(core::ModeValue{ inKeyData.mode == api::KeyMode::major ? core::ModeEnum::major : core::ModeEnum::minor });
+                }
+
+            } else {
+                key->getKeyChoice()->setChoice(core::KeyChoice::Choice::nonTraditionalKey);
+
+                for (const api::PitchData& p : inKeyData.nonTraditionalPitchData)
+                {
+                    core::NonTraditionalKeyPtr k = core::makeNonTraditionalKey();
+                    if (p.step == api::Step::count || p.step == api::Step::unspecified)
+                    {
+                        // Would you prefer throw? assert?
+                        continue;
+                    }
+
+                    auto s = std::make_shared<core::KeyStep>( myConverter.convert(p.step) );
+                    k->setKeyStep(s);
+
+                    if (p.accidental != api::Accidental::none)
+                    {
+                        k->setHasKeyAccidental(true);
+                        auto acc = std::make_shared<core::KeyAccidental>( myConverter.convert(p.accidental) );
+                        k->setKeyAccidental(acc);
+                    }
+
+                    if (p.alter != 0) {
+                        k->setKeyAlter(std::make_shared<core::KeyAlter>(core::Semitones{ double(p.alter) }));
```

Comment:
Drops support for weird semitones :/ Maybe should use doubles in "NonTraditionalData"?

Reply:
I think we need to adapt the `int alter` and `double cents` pattern wherever we convert to/from `alter double`. See #50
Probably what we should do is move the calculations that I did in #50 to the MxConverter thing and use it in both places.

Reply:
And by using a custom data struct instead of pitchdata, I'll use a double so it doesn't conflict.